src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -0,0 +1,198 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_APPLICANT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TITLE;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.applicant.Applicant;
+import seedu.address.model.listing.JobDescription;
+import seedu.address.model.listing.JobTitle;
+import seedu.address.model.listing.Listing;
+
+/**
+ * Edits a listing in the listing book.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits a listing in the listing book. "
+            + "by the index number used in the displayed listing book. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TITLE + "TITLE] "
+            + "[" + PREFIX_DESCRIPTION + "DESCRIPTION] "
+            + "[" + PREFIX_APPLICANT + "APPLICANT]...\n"
+            + "[" + PREFIX_APPLICANT + "APPLICANT (optional)]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_TITLE + "Cool job title "
+            + PREFIX_APPLICANT + "John "
+            + PREFIX_APPLICANT + "Sam";
+
+    public static final String MESSAGE_EDIT_LISTING_SUCCESS = "Edited Listing: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_LISTING = "This listing already exists in the listing book";
```

Comment:
minor suggestion: instead of 
```
"This listing already exists in the listing book"
```
the error message should be like "Your edits don't change anything
```
"It seems that the changes you made did not modify the original version! 
Please review your edits and ensure that they are reflecting the intended changes!"
```